python-frontend/src/main/java/org/sonar/python/types/DeclaredType.java

```diff
@@ -149,10 +155,16 @@ public String typeName() {
   }
 
   Symbol getTypeClass() {
+    if (typeClass == null) {
+      return TypeShed.typeShedClass(builtinFullyQualifiedName);
+    }
     return typeClass;
   }
 
   public Set<Symbol> alternativeTypeSymbols() {
+    if (alternativeTypeSymbols == null) {
+      return resolveAlternativeSymbols(getTypeClass(), typeArgs);
```

Comment:
I guess here we could set `alternativeTypeSymbols = resolveAlternativeSymbols(getTypeClass(), typeArgs)` to avoid redoing the computation?
Though I think we discussed it and concluded it would be better to just recompute so that it would be a bit easier to manage in SonarLint?

Reply:
Yes, I did this to ease the "cache" invalidation. I think that the declared types we're now computing lazily don't have `type args ` so we will never execute line 166 for the time being.
However I believe it's cleaner to have it already done properly to avoid any inconsistency problem later. wdyt?

Reply:
Makes sense to me!